conda_lock/src_parser/__init__.py

```diff
@@ -113,6 +269,23 @@ class LockMeta(StrictModel):
         ...,
         description="paths to source files, relative to the parent directory of the lockfile",
     )
+    time_metadata: Optional[TimeMeta] = Field(
+        default=None, description="Metadata dealing with the time lockfile was created"
+    )
+    git_metadata: Optional[GitMeta] = Field(
+        default=None,
+        description=(
+            "Metadata dealing with the git repo the lockfile was created in and the user that created it"
+        ),
+    )
+    inputs_metadata: Optional[Dict[str, InputMeta]] = Field(
+        default=None,
+        description="Metadata dealing with the input files used to create the lockfile",
+    )
+    custom_metadata: Optional[Dict[str, str]] = Field(
```

Comment:
Suppose I want to include something like `environment.yml` which contains lists. Shall we make this a fully general dict?

(Needs `from typing import Any`)

```suggestion
    custom_metadata: Optional[Dict[Any, Any]] = Field(
```

Reply:
Ideally not.  It places a lot of extra strain on other implemations that make use of the file format like `micromamba`.  `Any` is FAR too broad a type.  I can potentially see value in a `Union[str, list[str]]`

Reply:
Good point. Ideally it would be some sort of recursive type like `T: Union[int, str, list[T], dict[str,T]]`, but Pydantic chokes on that already.

Note how `requirements` must be at least `list[Union[str, dict[str, list[str]]]]` due to `["pip", {"pip": ["pypi-dep"]}]`. Thus to parse most `requirements.txt` we'd need at least something like
```
dict[str, Union[str, list[Union[str, dict[str, list[str]]]]]]
```
(assuming I didn't make any mistakes).

Given the complexity, I've convinced myself that this should be a separate feature. But in this case, we should probably make it clear in the `--help` that metadata-jsons/metadata-yamls contents must be `dict[str,str]`.

Reply:
Do we want to support `dict[str, Union[str, int, float]]`?